src/cmem/ActivityControl/SilkActivityControl.tsx

```diff
@@ -167,38 +186,125 @@ export function useSilkActivityControl({
+                                msg: simplifiedElapsedTime(elapsedTimeSegmented(period), translateUnits, true) ,
+                                period,
+                            },
+                        };
+                    });
+                }, 1000);
             }
-            currentStatus.current = status
-            setActivityStatus(status)
+        },
+        [translateUnits]
+    );
+
+    const clearStatusTimerRefs = (timer?: "runningTime" | "waitingTime") => {
+        const clearRunningTimer = () => {
+            if (runningTimerRef.current) {
+                clearInterval(runningTimerRef.current);
+                runningTimerRef.current = undefined;
+            }
+        };
+
+        const clearWaitingTimer = () => {
+            if (waitingTimerRef.current) {
+                clearInterval(waitingTimerRef.current);
+                waitingTimerRef.current = undefined;
+            }
+        };
+
+        switch (timer) {
+            case "runningTime":
+                clearRunningTimer();
+                break;
+            case "waitingTime":
+                clearWaitingTimer();
+                break;
+            default:
+                clearRunningTimer();
+                clearWaitingTimer();
         }
-        registerForUpdates(updateActivityStatus)
-        return unregisterFromUpdates
-    },
+    };
+
+    const updateStatusTimers = React.useCallback((status: SilkActivityStatusProps | undefined) => {
+        if (status) {
+            switch (status.statusName) {
+                case "Waiting":
+                    tickForStatusTimers("waitingTime", status);
+                    break;
+                case "Running":
+                    clearStatusTimerRefs("waitingTime");
+                    tickForStatusTimers("runningTime", status);
+                    break;
+                default:
+                    setStatusTimers({ ...defaultStatusTimers });
+                    clearStatusTimerRefs();
+            }
+        }
+    }, []);
+
+    //incase of unmount reset status timers
+    React.useEffect(() => {
+        return clearStatusTimerRefs;
+    }, []);
+
+    // Register update function
+    useEffect(
+        () => {
+            const updateActivityStatus = (status: SilkActivityStatusProps | undefined) => {
+                if (status?.concreteStatus !== "Waiting") {
+                    setShowStartPrioritized(false);
+                } else if (executePrioritized) {
+                    // Show start prioritized button only-if the activity is still in Waiting status after 2s
+                    setTimeout(() => {
+                        if (currentStatus.current?.concreteStatus === "Waiting") {
+                            setShowStartPrioritized(true);
+                        }
+                    }, 2000);
+                }
+                currentStatus.current = status;
+                updateStatusTimers(status);
+                setActivityStatus(status);
```

Comment:
The state of the widget will be updated relatively often. There can be many status updated during the `Running` phase. Would it be possible to only set the start time once, when switching to Waiting or Running? See `activityControlLabel` for a similar use case with elapsed time where a `ElapsedDateTimeDisplay` element is created.